src/array/ops/image.rs

```diff
@@ -227,7 +259,6 @@ impl ImageArray {
 
         assert_eq!(result.height(), h);
         assert_eq!(result.width(), w);
-        assert_eq!(result.channels(), c);
```

Comment:
Lets keep this validation in

Reply:
This `DaftImageBuffer::channels()` method was duplicating the `ImageMode::num_channels()` logic, so I'd like to consolidate that if possible. We're also no longer pulling the channels from the `DaftImageBuffer` in `from_daft_image_buffer()` (we're pulling from `ImageMode::num_channels()` instead), so I don't see motivation for continuing to have a `DaftImageBuffer::channels()` API.